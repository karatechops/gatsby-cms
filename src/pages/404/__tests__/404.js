import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery } from 'gatsby';
import NotFoundPage from '../index';

// Data for SEO component
beforeEach(() => {
  useStaticQuery.mockReturnValueOnce({
    site: {
      siteMetadata: {
        title: 'My site',
        author: 'Tester',
      },
    },
  });
});

describe('NotFoundPage', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <NotFoundPage
          data={{
            site: {
              siteMetadata: {
                title: 'My Test Site',
              },
            },
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
