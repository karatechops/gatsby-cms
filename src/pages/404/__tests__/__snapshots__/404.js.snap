// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage renders correctly 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 cqQlmF"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "auto",
      "maxWidth": "100%",
      "minHeight": "100vh",
    }
  }
>
  <main
    className="StyledBox-sc-13pk1d4-0 gYrpnw"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 QIaXS"
    >
      <h1
        className="StyledHeading-sc-1rdh4aw-0 jowXmd"
      >
        Not Found
      </h1>
      <p
        className="StyledParagraph-tbetod-0 gJWCSY"
      >
        You just hit a route that doesn't exist... the sadness.
      </p>
    </div>
  </main>
  <footer
    className="StyledBox-sc-13pk1d4-0 cJOxqP"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 ldwCNw"
    >
      <span
        className="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        Â© 
        2019
        , Built with
         
        <a
          className="StyledAnchor-sc-1rp7lwl-0 llbwuF"
          href="https://grommet.io"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <span
            aria-label="love"
            role="img"
          >
            ðŸ’œ
          </span>
        </a>
      </span>
    </div>
  </footer>
</div>
`;
